<launch>

    <!-- SYSTEM MODEL -->
    <!-- vehicle parameters-->
    
    <param name="m"             type="double"   value="2.424" />
    <param name="rho"             type="double"   value="1.225" />    
    <param name="lr"            type="double"   value="0.1203" />
    <param name="lf"            type="double"   value="0.1377" />
    <param name="Cm0"            type="double"   value="10.1305" />
    <param name="Cm1"            type="double"   value="1.05294" />
    <param name="C0"            type="double"   value="3.68918" />
    <param name="C1"            type="double"   value="0.0306803" />
    <param name="Cd_A"            type="double"   value="-0.657645" />
    <param name="Caf"            type="double"   value="1.3958" />
    <param name="Car"            type="double"   value="1.6775" />
    <param name="Iz"            type="double"   value="0.02" />
    

    <!-- Variables limits-->
    <param name="max_vel"           type="double"   value="5.0" />
    <param name="min_vel"           type="double"   value="-5.0" />
    <param name="dutycycle_max"     type="double"   value="1.0" />
    <param name="dutycycle_min"     type="double"   value="-1.0" />
    <param name="steer_max"         type="double"   value="0.25" />
    <param name="steer_min"         type="double"   value="-0.25" />
    <param name="lat_e_max"         type="double"   value="0.4" />   <!-- like halfWidth-->
    <param name="orient_e_max"      type="double"   value="0.4" />


    <!-- PLANNER    -->
    <param name="slow_mode_th"  type="double"   value="0.6" />
    <param name="planning_mode" type="int"      value="2" />
    <param name="halfWidth"     type="double"   value="0.4" />
    <param name="trackShape"    type="str"      value="oval_iri" />


    <!--DYNAMIC OBSERVER-->
    <!--<node pkg="lpv_mpc"         type="stateEstimator.py"    name="state_estimator" output="screen">
        <param name="loopRate"  type="double"   value="100"	/>
    </node>-->


    <!-- CONTROL -->
    
    <node pkg="lpv_mpc"         type="controllerMain.py"    name="control" output="screen">
        <param name="vel_th"    type="double"   value="0.7" />
        <param name="N"         type="int"      value="5" />
        <param name="Hz"        type="double"   value="30" />
        <param name="vel_ref"   type="double"   value="0.3"/>
    </node>

</launch>
